tensorflow/core/grappler/optimizers/auto_mixed_precision.cc

```diff
@@ -56,6 +59,22 @@ const char kCastToFp16[] = "CastToFp16";
 const char kCastToBf16[] = "CastToBf16";
 const char kCastToFp32[] = "CastToFp32";
 
+#if TENSORFLOW_USE_ROCM
+std::vector<std::string> FP16SupportedDevices = 
+{
+    "906", 
+    "908", 
+    "90a"
+};
+
+bool HasEnhancedFP16ComputeSupport(std::pair<int, int> gpu_arch){
```

Comment:
perhaps expose this in the .h file, to eliminate the need to cut-n-paste codes from here in the `*_test.cc` file?

Reply:
I originally put it into auto_mixed_precision.h but then I was a getting multiple definitions compilation error in the CI. Did you mean the function HasEnhancedFP16ComputeSupport, the data FP16ComputeSupportedDevices, or both?

Reply:
just the declaration of `HasEnhancedFP16ComputeSupport` in the .h file
that should get rid of the multiple definitions compilation error as well

Reply:
That's right, thanks!